processor/batchprocessor/config.pkl

```diff
@@ -0,0 +1,58 @@
+@go.Package { name = "go.opentelemetry.io/collector/processor/batchprocessor" }
+module BatchProcessorConfig
+
+import "package://pkg.pkl-lang.org/pkl-go/pkl.golang@0.7.0#/go.pkl"
+import "package://pkg.pkl-lang.org/pkl-pantry/org.openapis.v3.contrib@1.0.3#/SchemaGenerator.pkl"
+
+class MapstructureField extends go.Field {
+  structTags {
+    ["mapstructure"] = "%{name}"
+  }
+}
+
+class Config {
+  /// The time after which a batch will be sent regardless of size.
+  @MapstructureField
+  timeout: Duration(isPositive) = 200.ms
+
+  /// The size of a batch which after hit, will trigger it to be sent.
+  /// When this is set to zero, the batch size is ignored and data will be sent immediately
+  /// subject to only send_batch_max_size.
+  //TODO: The comments above contain "\n" when the OpenAPI schema is generated. Can we remove them?
+  @MapstructureField
+  send_batch_size: UInt32 = 8192
+
+  /// SendBatchMaxSize is the maximum size of a batch. It must be larger than SendBatchSize.
+  /// Larger batches are split into smaller units.
+  /// Default value is 0, that means no maximum size.
+  //TODO: Add a check for send_batch_max_size must be greater or equal to send_batch_size
+  @MapstructureField
+  send_batch_max_size: UInt32 = 0
+
+  /// MetadataKeys is a list of client.Metadata keys that will be
+  /// used to form distinct batchers.  If this setting is empty,
+  /// a single batcher instance will be used.  When this setting
+  /// is not empty, one batcher will be used per distinct
+  /// combination of values for the listed metadata keys.
+  //TODO: Add a check for duplicate entries?
+  @MapstructureField
+  metadata_keys: Listing<String>
+
+  /// MetadataCardinalityLimit indicates the maximum number of
+  /// batcher instances that will be created through a distinct
+  /// combination of MetadataKeys.
+  @MapstructureField
+  metadata_cardinality_limit: UInt32 = 1000
+}
+
+//TODO: Thios creates a Go struct with an "output" field. It seems useless - can we not generate it?
+output = SchemaGenerator.generate(Config).output
```

Comment:
Here we are using Pkl's [OpenAPI schema generator](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/org.openapis.v3.contrib/current/SchemaGenerator/index.html#generate%28%29). To generate the schema, we can run this command:
```
cd processor/batchprocessor
pkl eval -f json config.pkl
```

Example output:

<details>
  <summary>OpenAPI schema</summary>
  
```json
{
  "type": "object",
  "title": "Config",
  "properties": {
    "timeout": {
      "type": "object",
      "title": "Duration",
      "description": "The time after which a batch will be sent regardless of size.",
      "properties": {
        "value": {
          "type": "number",
          "description": "The value of this duration.\n\nReturns [Int] if possible and [Float] otherwise.\n\nFacts:\n```\n1.min.value == 1\n2.2.h.value == 2.2\n```"
        },
        "isoString": {
          "type": "string",
          "description": "An [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Durations) representation of this duration.\n\nThe ISO representation has the format `[-]PT{hours}H{minutes}M{seconds}S`, where\n\n- negative durations have a leading minus sign,\n- `{hours}` and `{minutes}` are positive integers,\n- `{seconds}` is a positive integer or a positive decimal with decimal point (`.`),\n- zero-valued components are omitted, and\n- durations with length zero are represented as `PT0S`.\n\nFacts:\n```\n1.5.h.isoString == \"PT1H30M\"\n-1.5.h.isoString == \"-PT1H30M\"\n1.23.s.isoString == \"PT1.23S\"\n0.ms.isoString == \"PT0S\"\n```"
        },
        "unit": {
          "type": "string",
          "title": "DurationUnit",
          "description": "The unit of this duration.\n\nFacts:\n```\n1.min.unit == \"min\"\n2.2.h.unit == \"h\"\n```",
          "enum": [
            "ns",
            "us",
            "ms",
            "s",
            "min",
            "h",
            "d"
          ]
        },
        "isPositive": {
          "type": "boolean",
          "description": "Tells if this duration has a value of zero or greater.\n\nFacts:\n```\n2.min.isPositive\n!(-2.min.isPositive)\n```"
        }
      },
      "additionalProperties": false,
      "required": [
        "value",
        "isoString",
        "unit",
        "isPositive"
      ]
    },
    "send_batch_size": {
      "type": "integer",
      "description": "The size of a batch which after hit, will trigger it to be sent.\nWhen this is set to zero, the batch size is ignored and data will be sent immediately\nsubject to only send_batch_max_size.",
      "minimum": 0,
      "maximum": 4294967295
    },
    "send_batch_max_size": {
      "type": "integer",
      "description": "SendBatchMaxSize is the maximum size of a batch. It must be larger than SendBatchSize.\nLarger batches are split into smaller units.\nDefault value is 0, that means no maximum size.",
      "minimum": 0,
      "maximum": 4294967295
    },
    "metadata_keys": {
      "type": "array",
      "description": "MetadataKeys is a list of client.Metadata keys that will be\nused to form distinct batchers.  If this setting is empty,\na single batcher instance will be used.  When this setting\nis not empty, one batcher will be used per distinct\ncombination of values for the listed metadata keys.",
      "items": {
        "type": "string"
      }
    },
    "metadata_cardinality_limit": {
      "type": "integer",
      "description": "MetadataCardinalityLimit indicates the maximum number of\nbatcher instances that will be created through a distinct\ncombination of MetadataKeys.",
      "minimum": 0,
      "maximum": 4294967295
    }
  },
  "additionalProperties": false,
  "required": [
    "timeout",
    "send_batch_size",
    "send_batch_max_size",
    "metadata_keys",
    "metadata_cardinality_limit"
  ]
}
  ```
</details>